SPECS/golang/golang.spec

```diff
@@ -1,8 +1,8 @@
 %global bootstrap_compiler_version 1.19.12-1
 %global goroot          %{_libdir}/golang
 %global gopath          %{_datadir}/gocode
-%global ms_go_revision  1
-%global ms_go_buildid   20240111.3
+%global ms_go_revision  2
+%global ms_go_buildid   20240320.5
```

Comment:
Should we move to `20240321.6` which is the most recent release with updated submodules ?

Reply:
The submodule and patches are the same in source control, and the src tarball content is showing up as 100% identical according to kdiff3 on Windows.

We only ran 1.21.8-3 (and 1.22.1-3) to exercise our 1ES pipeline template migration backports. Using it is perfectly fine, but not necessary.